internal/persistence/json/database.go

```diff
@@ -129,3 +149,49 @@ func (db *Database) write() error {
 	}
 	return nil
 }
+
+func (db *Database) MigrateToV1(providers []provider.Provider) {
+	if db.data.Version == "1" {
+		return
+	}
+	db.data.Version = "1"
+	newRecords := []record{}
+	for _, rec := range db.data.Records {
```

Comment:
nit perhaps separate the inner body of that for loop in its own function, to simplify a bit + early return + less indentation?